src/main/java/nl/nn/testtool/web/api/MetadataApi.java

```diff
@@ -64,15 +64,20 @@ public class MetadataApi extends ApiBase {
 	public Response getMetadataList(@PathParam("storage") String storageName,
 									@QueryParam("metadataNames") List<String> metadataNames,
 									@DefaultValue("-1") @QueryParam("limit") int limit,
-									@DefaultValue("") @QueryParam("filterHeader") String filterHeader,
-									@DefaultValue("(.*)") @QueryParam("filter") String filterParam ,
+									@DefaultValue("") @QueryParam("filterHeader") List<String> filterHeaders,
+									@DefaultValue("(.*)") @QueryParam("filter") List<String> filterParams ,
 									@Context UriInfo uriInfo) {
 
 		List<String> searchValues = new ArrayList<>();
 		for(String field : metadataNames) {
-			if (filterHeader.equals(field)) {
-				searchValues.add(filterParam);
-			} else {
+			boolean changed = false;
+			for (int filterHeaderIndex = 0; filterHeaderIndex < filterHeaders.size(); filterHeaderIndex++) {
```

Comment:
Maybe use `for (String filterHeader : filterHeaders) {` instead if you don't need the index

Reply:
The fori is not avoidable in this situation because of how the method was designed. For now I opted to use the methods already in use like the storage.getmetadata() method

Reply:
I mend to replace the for loop with a for each loop.

Reply:
ah I see what you mean now. I'll change it to that. The index is still needed but I'll just get it using the indexOf method instead

Reply:
Sorry, I was mistaken. I didn't see that you needed the index to then get the value of another array. Then the previous code was fine. The `indexOf` searches through the array and checks each value, that takes more to compute.

Reply:
alright, I'll change it back in the next commit